django/db/models/fields/related.py

```diff
@@ -613,6 +613,21 @@ def resolve_related_fields(self):
             )
             to_field = (self.remote_field.model._meta.pk if to_field_name is None
                         else self.remote_field.model._meta.get_field(to_field_name))
+            if (
+                to_field_name is not None and to_field_name != 'id' and
```

Comment:
Are these two conditions really necessary? It feels like they could be removed.

Reply:
No, After replacing the condition with your suggestion `self.remoted_field.model._meta.concrete_model`, we don't need them anymore.